addons/product/models/product_template.py

```diff
@@ -922,6 +915,28 @@ def _get_mapped_attribute_names(self, parent_combination=None):
             for attribute_value in all_product_attribute_values
         }
 
+    def _filter_combinations_impossible_by_config(self, combination_tuples, ignore_no_variant=False):
```

Comment:
`_filter_combinations_impossible_by_config` and `_is_combination_possible_by_config` are similar on a lot of points.

So of course it is not possible to use `_is_combination_possible_by_config` in `_filter_combinations_impossible_by_config`, but maybe we could do the opposite ?
`_is_combination_possible_by_config` calls `_filter_combinations_impossible_by_config`.

By doing this, we do not have to duplicate the code.

---

I was thinking of something along the lines of:

```python
def _is_combination_possible_by_config(self, combination, ignore_no_variant=False):
    self.ensure_one()
    return bool(self._filter_combinations_impossible_by_config([combination], ignore_no_variant))
```
(Not exactly an expert on yields, so it may not work like this, but you get the idea).

And of course you put back all the comments from `_is_combination_possible_by_config` in `_filter_combinations_impossible_by_config` !

Reply:
@DavidFesquet Sure, it's always nice to reduce copy-pasted code :slightly_smiling_face: